### ISSUE TITLE
Prepare data to send to Querier

### ISSUE THREAD

**Issue by NGA-TRAN** [influxdata/influxdb_iox#3640]
__Use case:__
This is a task of #3637 that send requested data to the Querier. A request will look like:
```rust
struct IngesterQueryRequest {
  // we can lookup in memory by the table
  table: String,
  // only select the columns we care about
  columns: Vec<String>,
  // start time of the query
  min_time: i64,
  // end time of the query
  max_time: i64,
  // predicate for filtering
  predicate: Option<Predicate>,
  // optional to only return rows with 
  // a sequence number > than this
  greater_than_sequence_number: Option<u64>
}
```

which means:

1.  Ingester needs to send its in-memory data of the specified `columns` of the specified `table` but only data newer than the specified `greater_than_sequence_number`.
2. I am not sure if the `min_time` and `max_time` will be used here. Our data is not sorted in the Ingester's in-memeory, it will take time to extract data in between that range if we want to do so and I vote not to do so.
3. Similary, for now, the `predicate` won't be useful either because without statistics, we won't be able to prune any batches from the information of the `predicate`

Hence if we only stick to #1, data preparation is as follows:

Todo: add the design for this here

**Comment by NGA-TRAN** [comment]
@carols10cents : make it a stream then. I definitely have `schema`. If you touch this struct soon, feel free to `data` to stream and add the `schema` field

**Comment by carols10cents** [comment]
Yup, calling https://docs.rs/futures/0.3.21/futures/stream/fn.empty.html creates an empty stream.
